Tests/SonarQube.MSBuild.Tasks.UnitTests/TaskUtilitiesTests.cs

```diff
@@ -0,0 +1,140 @@
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Common;
+using System.Diagnostics;
+using System.IO;
+using TestUtilities;
+
+namespace SonarQube.MSBuild.Tasks.UnitTests
+{
+    [TestClass]
+    public class TaskUtilitiesTests
+    {
+        public TestContext TestContext { get; set; }
+
+        #region Tests
+
+        [TestMethod]
+        [TestCategory("Task")] // Regression test for bug http://jira.codehaus.org/browse/SONARMSBRU-11
+        public void TaskUtils_LoadConfig_RetryIfConfigLocked_ValueReturned()
+        {
+            // Arrange
+            string testFolder = TestUtils.CreateTestSpecificFolder(this.TestContext);
+            string configFile = CreateAnalysisConfig(testFolder);
+
+            TestLogger logger = new TestLogger();
+
+            AnalysisConfig result = null;
+            PerformOpOnLockedFile(configFile, () => result = TaskUtilities.TryGetConfig(testFolder, logger), shouldTimeoutReadingConfig: false);
+
+            Assert.IsNotNull(result, "Expecting the config to have been loaded");
+
+            AssertRetryAttempted(logger);
+            logger.AssertWarningsLogged(0);
+            logger.AssertErrorsLogged(0);
+        }
+
+        [TestMethod]
+        [TestCategory("Task")] // Regression test for bug http://jira.codehaus.org/browse/SONARMSBRU-11
+        public void TaskUtils_LoadConfig_TimeoutIfConfigLocked_NullReturned()
+        {
+            // Arrange
+            // We'll lock the file and sleep for long enough for the task to timeout
+            string testFolder = TestUtils.CreateTestSpecificFolder(this.TestContext);
+            string configFile = CreateAnalysisConfig(testFolder);
+
+            TestLogger logger = new TestLogger();
+
+            AnalysisConfig result = null;
+
+            PerformOpOnLockedFile(configFile, () => result = TaskUtilities.TryGetConfig(testFolder, logger), shouldTimeoutReadingConfig: true);
+
+            Assert.IsNull(result, "Not expecting the config to be retrieved");
+
+            AssertRetryAttempted(logger);
+            logger.AssertWarningsLogged(0);
+            logger.AssertErrorsLogged(1);
+        }
+
+        #endregion
+
+        #region Public test helpers
+
+        /// <summary>
+        /// Performs the specified operation against a locked analysis config file
+        /// </summary>
+        /// <param name="configFile">The config file to be read</param>
+        /// <param name="op">The test operation to perform against the locked file</param>
+        /// <param name="shouldTimeoutReadingConfig">When the operation should timeout or not</param>
+        public static void PerformOpOnLockedFile(string configFile, System.Action op, bool shouldTimeoutReadingConfig)
```

Comment:
I've made this a public static so it can be called from the "IsTestByName" test class. I'll probably move it to a shared helper class in a later commit.